checkov/common/bridgecrew/integration_features/features/vulnerabilities_integration.py

```diff
@@ -0,0 +1,186 @@
+                                self.is_deeper_or_equal_level(sca_file_path, code_file_path)]
+
+            if not len(filtered_entries):
+                continue
+
+            # Create map with the relevant structure for the enrichment step
+            sast_files_by_packages_map = self.create_file_by_package_map(filtered_entries)
+
+            # Enrich the CVEs
+            self.enrich_cves_with_sast_data(current_cves, sast_files_by_packages_map)
+
+    '''
+    Each SCA report check has file_path, we want to getter same file_path so we won't have to calculate SAST language more then once
+    '''
+
+    def group_cves_checks_by_files(self, cves_checks: List[Record]) -> Dict[str, List[Record]]:
+        # Create a dictionary to store the grouped records
+        grouped_records: Dict[str, List[Record]] = defaultdict()
+
+        # Group the records by the 'file_path' key
+        for record in cves_checks:
+            file_path = record.file_path
+            if file_path not in grouped_records:
+                grouped_records[file_path] = []
+            grouped_records[file_path].append(record)
+
+        return grouped_records
+
+    '''
+    convert SAST report structure to a sturcture grouped by package_name, for better performance in the enrich step
+    '''
+
+    def create_file_by_package_map(self, filtered_entries: List[Tuple[Any, Any]]) -> Dict[str, List[str]]:
+        sast_files_by_packages_map: Dict[str, List[str]] = defaultdict()
+        for code_file_path, sast_data in filtered_entries:
+            for package_name in sast_data['all']:
+                clean_package_name = package_name.strip("'")
+                # Normalize package name
+                normalize_package_name = self.normalize_package_name(clean_package_name)
+
+                if clean_package_name not in sast_files_by_packages_map:
+                    sast_files_by_packages_map[clean_package_name] = []
+                if normalize_package_name not in sast_files_by_packages_map:
+                    sast_files_by_packages_map[normalize_package_name] = []
+
+                sast_files_by_packages_map[clean_package_name].append(code_file_path)
+                sast_files_by_packages_map[normalize_package_name].append(code_file_path)
+
+        return sast_files_by_packages_map
+
+    '''
+    enrich each CVE with the risk factor of IsUsed - which means there is a file the use the package of that CVE
+    '''
+
+    def enrich_cves_with_sast_data(self, current_cves: List[Record], sast_files_by_packages_map: Dict[str, List[str]]) -> None:
+        for cve_check in current_cves:
+            if cve_check.vulnerability_details:
+                package_name = cve_check.vulnerability_details.get('package_name', '')
+                normalize_package_name = self.normalize_package_name(package_name)
+                cve_check.vulnerability_details.get('risk_factors', {})['IsUsed'] = False
```

Comment:
do we have an enum for package data fields? if so let's use them, and if not let's create it :)

Reply:
it a change in many places, we can do this refactor in a separate pr